x-pack/plugins/ml/public/application/trained_models/models_management/start_deployment_setup.tsx

```diff
@@ -0,0 +1,226 @@
+            defaultMessage="Number of allocations"
+          />
+        }
+        hasChildLabel={false}
+        helpText={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsHelp"
+            defaultMessage="Each allocation means the model gets another thread for executing parallel inference requests"
+          />
+        }
+      >
+        <EuiFieldNumber
+          min={1}
+          name={'numOfAllocations'}
+          value={numOfAllocation}
+          onChange={(event) => {
+            onConfigChange({ ...config, numOfAllocations: Number(event.target.value) });
+          }}
+        />
+      </EuiFormRow>
+
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationLabel"
+            defaultMessage="Threads per allocation"
+          />
+        }
+        hasChildLabel={false}
+        helpText={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationHelp"
+            defaultMessage="Each allocation is may be using a number of threads to parallelize each individual inference request"
+          />
+        }
+      >
+        <EuiButtonGroup
+          legend={i18n.translate(
+            'xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationLegend',
+            {
+              defaultMessage: 'Threads per allocation selector',
+            }
+          )}
+          name={'threadsPerAllocation'}
+          idSelected={toggleIdSelected}
+          onChange={(optionId) => {
+            const value = threadsPerAllocationsOptions.find((v) => v.id === optionId)!.value;
+            onConfigChange({ ...config, threadsPerAllocations: value });
+          }}
+          options={threadsPerAllocationsOptions}
+        />
+      </EuiFormRow>
+    </EuiForm>
+  );
+};
+
+interface StartDeploymentModalProps {
+  modelId: string;
+  onConfigChange: (config: ThreadingParams) => void;
+  onClose: () => void;
+}
+
+/**
+ * Modal window wrapper for {@link StartDeploymentSetup}
+ *
+ * @param onConfigChange
+ * @param onClose
+ */
+export const StartDeploymentModal: FC<StartDeploymentModalProps> = ({
+  modelId,
+  onConfigChange,
+  onClose,
+}) => {
+  const [config, setConfig] = useState<ThreadingParams>({
+    numOfAllocations: 1,
+    threadsPerAllocations: 1,
+  });
+
+  return (
+    <EuiModal onClose={onClose} initialFocus="[name=numOfAllocations]">
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <h2>
+            <FormattedMessage
+              id="xpack.ml.trainedModels.modelsList.startDeployment.modalTitle"
+              defaultMessage="Start {modelId} deployment"
+              values={{ modelId }}
+            />
+          </h2>
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
```

Comment:
I wonder if it's worth adding some extra text to this modal to explain the benefit of these settings, in a callout or info tooltip perhaps, with content similar to that used in the docs. @szabosteve @lcawl any thoughts?

```
TIP: You can set additional deployment options that affect the performance of
the model. For example, specify the number of allocations and the threads per
allocation to optimize the throughput and speed of inference requests in your
production environment. For more information about these options, refer to the
{ref}/start-trained-model-deployment.html[start trained model deployment API].
```